cubic-server/protocol/ClientPackets.cpp

```diff
@@ -23,6 +23,53 @@ std::shared_ptr<std::vector<uint8_t>> protocol::createStatusResponse(const Statu
     return packet;
 }
 
+std::shared_ptr<std::vector<uint8_t>> protocol::createLoginSuccess(const LoginSuccess &in)
+{
+    std::vector<uint8_t> payload;
+    serialize(payload, in.uuid, addUUID,
+        in.username, addString,
+        in.numberOfProperties, addVarInt,
+        in.name, addString,
+        in.value, addString,
+        in.isSigned, addBoolean);
+    if (in.isSigned)
+        serialize(payload, in.signature.value(), addString);
+    auto packet = std::make_shared<std::vector<uint8_t>>();
+    finalize(*packet.get(), payload, (int32_t)ClientPacketID::LoginSuccess);
+    return packet;
+}
+
+std::shared_ptr<std::vector<uint8_t>> protocol::createLoginPlay(const LoginPlay &in)
+{
+    std::vector<uint8_t> payload;
+    serialize(payload,
+        in.entityID, addVarInt,
+        in.isHardcore, addBoolean,
+        in.gamemode, addByte,
+        in.previousGamemode, addByte,
+        in.dimensionCount, addVarInt,
+        in.dimensionNames, addArray<std::string, addString>,
```

Comment:
even if it's in the package ?! Insane if it's the case !
![image](https://user-images.githubusercontent.com/60519602/208811365-ee3f4cfb-9a49-45ed-8192-5fcadaa9dc24.png)

Reply:
Yes, the `addArray` method add a varint before corresponding on the `std::vector` size.
So there is no real need to hold the size of a specific array, because `std::vector` do that for us